patch-package
--- a/node_modules/join-monster/dist/query-ast-to-sql-ast/index.js
+++ b/node_modules/join-monster/dist/query-ast-to-sql-ast/index.js
@@ -242,6 +242,11 @@ function handleTable(sqlASTNode, queryASTNode, field, gqlType, namespace, grabMa
     sqlASTNode.limit = (0, _util.unthunk)(field.limit, sqlASTNode.args || {}, context);
   }
 
+  if (field.offset) {
+    (0, _assert2.default)(field.limit, '`limit` is required with `offset`');
+    sqlASTNode.offset = (0, _util.unthunk)(field.offset, sqlASTNode.args || {}, context);
+  }
+
   if (sqlASTNode.paginate) {
     getSortColumns(field, sqlASTNode, context);
   }
--- a/node_modules/join-monster/dist/stringifiers/shared.js
+++ b/node_modules/join-monster/dist/stringifiers/shared.js
@@ -117,7 +117,7 @@ function interpretForOffsetPaging(node, dialect) {
   }
 
   let limit = ['mariadb', 'mysql', 'oracle'].includes(name) ? '18446744073709551615' : 'ALL';
-  let offset = 0;
+  let offset = node.offset || 0;
   if ((_ref4 = node) != null ? (_ref4 = _ref4.args) != null ? _ref4.first : _ref4 : _ref4) {
     limit = parseInt(node.args.first, 10);
 
